torch/csrc/jit/codegen/cuda/index_compute.cpp

```diff
@@ -462,12 +463,18 @@ void IndexCompute::handle(Merge* merge) {
     index_map_[inner_id] = zero;
 
     extent_map_[outer_id] = getExtent(out_id);
+    if (hasZeroMerged(out_id)) {
+      zero_merged_in_.insert(outer_id);
+    }
   } else if (outer_id->isBroadcast() && outer_extent->isOneInt()) {
     // Propagate away from broadcast dims
     index_map_[outer_id] = zero;
     index_map_[inner_id] = out_ind;
 
     extent_map_[inner_id] = getExtent(out_id);
+    if (hasZeroMerged(out_id)) {
```

Comment:
This is subtle difference from the previous semantic. I think this was dead path before since reference replay would never replay any id as broadcast. Would need to revisit and clean up further for more unified logic.